packages/devtools_app/test/inspector_layout_tab_test.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:devtools_app/src/inspector/flutter/layout_tab.dart';
+import 'package:devtools_app/src/ui/fake_flutter/_real_flutter.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('FlexProperties.fromJson creates correct value from enum', () {
+    for (var direction in Axis.values) {
+      for (var mainAxisAlignment in MainAxisAlignment.values) {
+        for (var mainAxisSize in MainAxisSize.values) {
+          for (var crossAxisAlignment in CrossAxisAlignment.values) {
+            for (var textDirection in TextDirection.values) {
+              for (var verticalDirection in VerticalDirection.values) {
+                for (var textBaseline in TextBaseline.values) {
+                  final Map<String, Object> flexJson = {
+                    'direction': direction.toString(),
+                    'mainAxisAlignment': mainAxisAlignment.toString(),
```

Comment:
these enums are unlikely to be changed in package:flutter without an incredibly painful breaking change so if you want to save a few bytes you could serialize them as
mainAxisAlignment.index
The other option is to serialize based on 
stripping out the enum name so you serialize
MainAxisAlignment.start as 'start'
See EnumProperty in EnumProperty in Flutter that does this.
In fact, consider reusing the actual DiagnosticsNode property serialization for all these enums and doubles.
I expect you will already see all of these properties serialized as EnumProperty and IntProperty in debugFillProperties method for the related RenderObject.
You could deserialize them out of the RemoteDiagnosticsNode objects into the FlexProperties data structure.
The one thing it would give you is an indication of which properties are considered to be at their default value.
The json will be a bit less compact than what you have but I expect it wouldn't be a performance blocker.
I'm not sure which approach will be better so lets chat about it tomorrow. There are pros and cons of both approaches.

Reply:
Alright, let's chat about this more this morning!